### ISSUE TITLE
DETR for Multi-Modal Trajectory Prediction

### ISSUE THREAD

**Issue by smartbarbarian** [sshaoshuai/MTR#24]
I have attempted to predict trajectories using DETR and the hard-assignment strategy, without incorporating auxiliary loss. Unfortunately, this approach consistently results in straight, linear trajectories for each sample provided. I observed that your method is similar but includes auxiliary loss. I'm interested in understanding if the inclusion of this additional loss component effectively mitigates the collapsing problem. I would greatly appreciate any advice or suggestions you can provide on this matter. Thank you so much!

**Comment by sshaoshuai** [comment]
I don't think the auxiliary loss will have a great effect for MTR framework. From Table 4 of our paper (https://arxiv.org/pdf/2209.13508.pdf), you may see that MTR without Dense Future Prediction can also achieve promising performance. 
Instead, the design of intention queries with intention points is the key of MTR framework.

**Comment by smartbarbarian** [comment]
DETR trains the model by using auxiliary loss from every decoder layer, while I only used the loss from the last layer. This approach initially led to a collapsing issue with the hard-assignment strategy. However, by clustering the trajectories and applying the hard-assignment strategy to the clustered ones, the model worked. Furthermore, I am intrigued as to why you chose NMS over the clustering approach, especially considering that different objects require distinct thresholds for NMS.

**Comment by smartbarbarian** [comment]
For dynamic search queries, you predicted the trajectories and utilized the last points. However, it appears that an MLP could directly replace this step. The design seems to primarily incorporate the inductive bias. Similarly, static intention queries appear to be learnable with sufficient training.

**Comment by smartbarbarian** [comment]
Hi @sshaoshuai ,
Apologies for my delayed response. Could you please consider reopening the issue?  I noticed that the authors of DAB-DETR employed a mixed query selection in their subsequent paper, DINO, as opposed to the more complex anchor design used in DAB-DETR. Have you experimented with other anchor and content query designs?
Best,

**Comment by sshaoshuai** [comment]
Thanks for your interest. Regarding your questions, 
1. NMS is simple and straightforward so I use it. Clustering algorithm seems also reasonable, but my previous experiments show that it is hard to tune since there may also be some hyper-parameters
2. Both the dynamic queries and static queries actually are intention queries with some explicit points, which is the key of MTR (of course there may be other alternatives with small modifications that can achieve similar performance). 
3. Replacing static intention queries with learnable queries actually achieves worse mAP performance, which is similar to the comparison of MTR/MTR-e2e in our paper
4. I did not remember clearly what other designs I have tried in MTR framework, as I think it is not the focus of MTR framework. I think there may be some small improvements by adopting better anchor/content query designs, but I do not think it will change too much. Since again, the key of MTR is the explicit intention queries.
